src/Illuminate/Validation/Rules/Dimensions.php

```diff
@@ -113,65 +264,205 @@ public function maxHeight($value)
+                $value = $this->{$property};
+
+                if (is_array($value)) {
+                    $value = implode(',', $value);
+                }
+
+                $ruleName = Str::snake($property);
+                $rules[] = "{$ruleName}:{$value}";
+            }
+        }
+
+        return array_merge($rules, $this->customRules);
+    }
+
+    /**
+     * Set the default callback to be used for determining default rules.
+     *
+     * If no arguments are passed, the default dimensions rule configuration will be returned.
      *
-     * @return string
+     * @param  static|callable|null  $callback
+     * @return static|void
      */
-    public function __toString()
+    public static function defaults($callback = null)
+    {
+        if (is_null($callback)) {
+            return static::default();
+        }
+
+        if (! is_callable($callback) && ! $callback instanceof static) {
+            throw new InvalidArgumentException('The given callback should be callable or an instance of '.static::class);
+        }
+
+        static::$defaultCallback = $callback;
+    }
+
+    /**
+     * Get the default configuration of the dimensions rule.
+     *
+     * @return static
+     */
+    public static function default()
+    {
+        $dimensions = is_callable(static::$defaultCallback)
+            ? call_user_func(static::$defaultCallback)
+            : static::$defaultCallback;
+
+        return $dimensions instanceof Rule ? $dimensions : new self;
+    }
+
+    /**
+     * Specify additional validation rules that should be merged with the default rules during validation.
+     *
+     * @param  string|array  $rules
+     * @return $this
+     */
+    public function rules($rules)
+    {
+        $this->customRules = array_merge($this->customRules, Arr::wrap($rules));
+
+        return $this;
+    }
+
+    /**
+     * Determine if the validation rule passes.
+     *
+     * @param  string  $attribute
+     * @param  mixed  $value
+     * @return bool
+     * */
+    public function passes($attribute, $value)
     {
-        $result = '';
+        $this->messages = [];
+
+        $validator = Validator::make(
+            $this->data,
+            [$attribute => $this->buildValidationRules()],
+            $this->validator->customMessages,
+            $this->validator->customAttributes
+        );
 
-        foreach ($this->constraints as $key => $value) {
-            $result .= "$key=$value,";
+        if ($validator->fails()) {
+            return $this->fail($validator->messages()->all());
         }
 
-        return 'dimensions:'.substr($result, 0, -1);
+        return true;
+    }
+
+    /**
+     * Adds the given failures, and return false.
+     *
+     * @param  array|string  $messages
+     * @return bool
+     */
+    protected function fail($messages)
+    {
+        $messages = collect(Arr::wrap($messages))->map(
+            fn ($message) => $this->validator->getTranslator()->get($message)
```

Comment:
I wonder if some variables/properties in this rule can be typed better 🤔 
```suggestion
     * @param  string[]|string  $messages
     * @return bool
     */
    protected function fail($messages)
    {
        $messages = collect(Arr::wrap($messages))->map(
            fn (string $message) => $this->validator->getTranslator()->get($message)
```